cf-net/cf-net.c

```diff
@@ -0,0 +1,507 @@
+                ManPageWrite(out, "cf-net", time(NULL),
+                             CF_NET_SHORT_DESCRIPTION,
+                             CF_NET_MANPAGE_LONG_DESCRIPTION,
+                             OPTIONS, HINTS,
+                             true);
+                FileWriterDetach(out);
+                exit(EXIT_SUCCESS);
+                break;
+            }
+            default:
+            {
+                break;
+            }
+        }
+    }
+    (*cmd) = argv[optind];
+    (*args) = &(argv[optind+1]);
+    return 0;
+}
+
+// Just for testing argument parsing
+int CFNetPrint(char **args)
+{
+    for (int i = 0; args[i] != NULL; ++i)
+    {
+        printf("%s ", args[i]);
+    }
+    printf("\n");
+    return 0;
+}
+
+void CFNetInit()
+{
+    CryptoInitialize();
+    LoadSecretKeys();
+    cfnet_init(NULL, NULL);
+}
+
+AgentConnection* CFNetOpenConnection(char *server)
+{
+
+    AgentConnection *conn = NULL;
+    {
+        ConnectionFlags connflags = {
+            .protocol_version = CF_PROTOCOL_LATEST,
+            .trust_server = true
+        };
+        int err = 0;
+        char *buf = xstrdup(server);
+        char *host, *port;
+        ParseHostPort(buf, &host, &port);
+        if (port == NULL)
+        {
+            port = CFENGINE_PORT_STR;
+        }
+        conn = ServerConnection(host, port, 30, connflags, &err);
+        free(buf);
+    }
+    if (conn == NULL)
+    {
+        nprint("Failed to connect to '%s'.\n", server);
+        return NULL;
+    }
+    return conn;
+}
+
+int CFNetConnectSingle(char *server)
+{
+    AgentConnection *conn = CFNetOpenConnection(server);
+    if(conn == NULL)
+    {
+        return -1;
+    }
+    nprint("Connected & authenticated successfully to '%s'.\n", server);
+    DeleteAgentConn(conn);
+    return 0;
+}
+
+int CFNetConnect(char **args)
+{
+    for (int i = 0; args[i] != NULL; ++i)
+    {
+        cfnet_silent = false;
+        CFNetConnectSingle(args[i]);
+        cfnet_silent = true;
+    }
+    return 0;
+}
+
+void CFNetStatPrint(const char* file, int st_mode)
+{
+    if (S_ISDIR(st_mode))
+    {
+        nprint("'%s' is a directory.\n", file);
+    }
+    else if (S_ISREG(st_mode))
+    {
+        nprint("'%s' is a regular file.\n", file);
+    }
+    else if (S_ISSOCK(st_mode))
+    {
+        nprint("'%s' is a socket.\n", file);
+    }
+    else if (S_ISCHR(st_mode))
+    {
+        nprint("'%s' is a character device file.\n", file);
+    }
+    else if (S_ISBLK(st_mode))
+    {
+        nprint("'%s' is a block device file.\n", file);
+    }
+    else if (S_ISFIFO(st_mode))
+    {
+        nprint("'%s' is a named pipe (FIFO).\n", file);
+    }
+    else if (S_ISLNK(st_mode))
+    {
+        nprint("'%s' is a symoblic link.\n", file);
+    }
+    else
+    {
+        nprint("'%s' has an unrecognized st_mode.\n", file);
+    }
+}
+
+int CFNetStat(char **args)
+{
+    char *server = args[0];
+    char *file = args[1];
```

Comment:
The server should be an optional argument (`-H` like cf-runagent) and default to policy hub if not present. See CFE-2493 for description and let me know if you disagree or not understand why, since there are reasons for that. I'm open to discussing syntax and commands and everything else, but please keep the discussion in the ticket.